### DISCUSSION TITLE
Patch increase and decrease

### DISCUSSION THREAD

**Discussion by pkfrulz** [GeekFunkLabs/fluidpatcher#93] [General]
I'm using squish box latest version with 16x2 display and rotary encoder. I have a small midi keyboard (WORLDE Panda MINI). I want to increase and decrease patches using program (+,-) buttons. when I press program  + button it increases the patch and I can hear the sound of patch that changed . but in lcd screen its not changed. when I use rotary encoder there's no any problem the patches can increase and decrease and the lcd screen updates normally and works fine. but I have the problem with my midi keyboard program increase, decrease button. it changes the patches but the lcd screen is not updated. and I programmed my midi keyboard pads to cc 21,22 momentary but that's not working ! and I checked my midi keyboard signal on midi monitor. its sending the correct cc numbers in correct midi channel (channel 1) as momentary. can anyone help me in this

**Comment by albedozero** [comment]
There was a router rules bug, fixed by #95, that should remedy the patch change rules not working.

When you send a program change message with your keyboard, fluidsynth changes the preset in the sound font that is assigned to that channel. This is different from the patches that you create in your bank files. If you save the patch, it will be saved with the new preset you selected.

**Reply by pkfrulz** [reply]
thanks for the reply. I will test it
